src/main.rs

```diff
@@ -21,9 +22,38 @@ fn main() -> Result<(), Box<dyn Error>> {
     let host = matches.get_one::<String>("Broker").unwrap();
     let port = *matches.get_one::<u16>("Port").unwrap();
 
-    let client_id = format!("mqttui-{:x}", rand::random::<u32>());
+    let client_id = matches
+        .get_one::<String>("ClientId")
+        .cloned()
+        .unwrap_or_else(|| format!("mqttui-{:x}", rand::random::<u32>()));
+
+    let encryption = matches.get_one::<bool>("Encryption").copied();
+    let encryption = match (encryption, port) {
+        (Some(encryption), _) => encryption,
+        (None, 8883) => true,
+        _ => false,
+    };
```

Comment:
I am not sure why this part exists. Why not default to encryption true on any port and have a `--no-encryption` flag?

Thinking already about the WebSocket logic with Urls we probably need our own `Url` → `MqttOptions` logic which also handles encryption usefully then?
`mqtt://` and `mqtts://` could specify this then instead of an `--encryption` flag?

Reply:
This way does not break existing useage of the tool much. And 8883 is a standard port for MQTT with tls, so that exception for a default makes sense. I mean we could change to have encryption on every port except 1883 if nothing is specifed. 

the only thing I do not like about the URL is that it does not support binary passwords afaik.

Reply:
Yeah `Url ` --> `MqttOptions` logic will be needed in any case. But that would change quite a bit of stuff more then just the encryption logic, I am unsure how well rumqttc handles the different cases with encryption for websockets anyway. 

Kinda feels a bit out of scope for current PR. I like to have as much autonomy in the CLI tool. instead of removing this part

I would rather have us add the corresponding port logic, for example, choose port 8883 if encryption is set to true, but no port has been specified.

Reply:
Yeah, definitely. The `Url` → `MqttOptions` stuff is something for the future, I just wanted to think about it already with this PR to make sure we dont create a huge blocker with this.

I didnt realized the `1883` vs `8883` Port difference. Thanks for the explanation there!

Also yeah, lets have this non breaking and keep it simpler.